GitUI/Theming/ThemeEditor.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+using GitCommands;
+using GitExtUtils.GitUI.Theming;
+
+namespace GitUI.Theming
+{
+    public class ThemeEditor
```

Comment:
I'm struggling grasping the purpose of this class. I'm finding the name confusing, and it is overloaded with different concerns - reading/writing files, showing dialogs to the user, setting/reading colours etc. This makes the functionality quite incoherent.
We had similar problems dealing with repository histories (look at https://github.com/gitextensions/gitextensions/pull/4766) 
Can we break this apart into SOLID classes?

Reply:
There is `FormThemeEditor` which presents UI to save / load color scheme or modify individual colors. `ThemeEditor` is the part which does not depend on the `Form` itself.

It is used to handle `FormThemeEditor` button clicks, and besides that, it can be used to access `FormThemeEditor`'s functionality without creating a form. For instance, when loading current theme at application startup.

Maybe we should consider removing direct access from `FormThemeEditor` to `ThemeManager`
by proxying it through `ThemeEditor` to emphasice the role of `ThemeEditor` as handler of all 'business logic' of `FormThemeEditor`

Reply:
> It is used to handle `FormThemeEditor` button clicks, and besides that, it can be used to access `FormThemeEditor`'s functionality without creating a form

We are calling such classes "controllers", e.g. `FormBrowseController`, `FormFileHistoryController`, etc.

> Maybe we should consider removing direct access from `FormThemeEditor` to `ThemeManager`
> by proxying it through `ThemeEditor` to emphasize the role of `ThemeEditor` as handler of all 'business logic' of `FormThemeEditor`

That's a good idea!

Reply:
done